xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -2006,63 +2003,67 @@ struct AppModel: ModelProtocol {
     }
     
     /// Index a sphere to the database
-    static func indexPeer(
+    static func indexPeers(
         state: Self,
         environment: Environment,
-        petname: Petname
+        petnames: [Petname]
     ) -> Update<Self> {
-        logger.log(
-            "Indexing peer",
-            metadata: [
-                "petname": petname.description
-            ]
-        )
+        
         let fx: Fx<Action> = Future.detached(priority: .background) {
-            do {
-                let peer = try await environment.data.indexPeer(
-                    petname: petname
-                )
-                return Action.succeedIndexPeer(peer)
-            } catch {
-                return Action.failIndexPeer(
-                    petname: petname,
-                    error: error
+            var success: [PeerRecord] = []
+            var fail: [Petname] = []
+            
+            for petname in petnames {
+                logger.log(
+                    "Indexing peer",
+                    metadata: [
+                        "petname": petname.description
+                    ]
                 )
+                
+                do {
+                    let peer = try await environment.data.indexPeer(
+                        petname: petname
+                    )
+                    success.append(peer)
+                    await Task.yield()
+                } catch {
+                    fail.append(petname)
+                }
             }
```

Comment:
Running indexing serially instead of concurrently. We should factor this into a method.

Reply:
I managed to add some tests to assert concurrency behaviour around yielding, but I can't think of a way to _actually_ index a peer without a gateway in the mix.

It might be possible if we have mocked versions of every service `DataService` calls out to but... I'm not a huge fan of writing 100s of lines of mocks which can just as easily introduce problems.